.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,38 @@
+name: ci hive docker image
+
+on:
+  push:
+    branches:
+      - "master"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        
+      -
+        name: Build and push
+        uses: docker/build-push-action@v4
+        with:
+          context: ./packaging/src/docker/
+          file: ./packaging/src/docker/Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/hive:test-image
```

Comment:
We'd better add the real version to the image instead of `test-image`, and I'm thinking it would be great if we can determine the `HADOOP_VERSION`, `TEZ_VERSION` from the project.

Reply:
How about -hive:dev or -hive:daily?
The GA version should be the same as the industry follows like hive4.0-latest imho

Reply:
For GA:  the versions that are set in the .yml file were manually configured after looking at the hive/pom.xml file.

For hive:daily, i think we can obtain them from the pom.xml file

Reply:
Can we trigger the build for GA automatically?
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release
I think it makes some troubles every time we should add the new GA build manually, it adds extra steps for releasing the new version, sometimes we may even forget about it.
For the old released version, I think we can push the image manually.

Reply:
> Can we trigger the build for GA automatically?
>https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release

Since we are pushing the changes to the dockerhub for the 1st time, we wanted to trigger it with workflow_dispatcher to verify the dickerhub integration.

Once we verify that this GH action succeeds, we can set it to trigger automatically and update all the images on every release or once every three months.

> I think it makes some troubles every time we should add the new GA build manually, it adds extra steps for
> releasing the new version, sometimes we may even forget about it.

I agree, but I think for a new GA :
- We will not have prior knowledge of the versions of hive, tez and hadoop to use in the next GA. (Workaround could be: obtain from pom.xml)
- Someone will have to build the new GA docker images locally and verify if it's working before we push them to docker hub.

So that is why i was thinking we should retain the manual step at release time. 
Other repos follow something similar: https://github.com/apache/spark-docker/tree/master/.github/workflows